.toolversions

```diff
@@ -0,0 +1,20 @@
+cmake="
+  DeclaredVersion='3.6.0'
+  SearchPathsWindows='%programfiles(x86)%\\CMake\\bin\\cmake.exe
+                      %programfiles%\\CMake\\bin\\cmake.exe
+                      %programfiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\IDE\\CommonExtensions\\Microsoft\\CMake\\CMake\\bin\\cmake.exe
+                      %programfiles(x86)%\\Microsoft Visual Studio\\2017\\Professional\\Common7\\IDE\\CommonExtensions\\Microsoft\\CMake\\CMake\\bin\\cmake.exe'
+  SearchPathOSXTools='Tools/downloads/CMake/cmake-3.6.0-Darwin-x86_64/CMake.app/Contents/bin/cmake'
```

Comment:
>SearchPathOSXTools [](start = 2, length = 18)

The `Tools` suffix seems out of place.  I would expect the platform to be the last segment just like the other properties.  Maybe a name that better describes this path versus the machine wide path would be better.  Is `Local` better?  Additionally it seems like the Tools/downloads/cmake portion of this path should be specified separately.  This separate property should be used for search as well as the location to install the tool too if not found.  I would also not assume that location is part of the repo.  Additionally I would consider this location property to be optional, if not specified we would default it to Tools\downloads\<toolname>